packages/cadl-ranch-specs/http/type/model/error/mockapi.ts

```diff
@@ -0,0 +1,37 @@
+import { passOnSuccess, mockapi, json, passOnCode } from "@azure-tools/cadl-ranch-api";
+import { ScenarioMockApi } from "@azure-tools/cadl-ranch-api";
+
+export const Scenarios: Record<string, ScenarioMockApi> = {};
+
+Scenarios.Type_Model_Error_Single_validResponse = passOnSuccess(
+  mockapi.get("/type/model/error/single/valid", (req) => {
+    return { status: 200, body: json({ username: "jane", password: "doe" }) };
+  }),
+);
+
+Scenarios.Type_Model_Error_Single_invalidResponse = passOnCode(
+  304,
+  mockapi.get("/type/model/error/single/invalid", (req) => {
+    return { status: 304, body: json({ code: "Not Modified" }) };
```

Comment:
I wouldn't do 304, this will trigger Azure-Core behaviors on redirection, and you may never see the response at all, or a weird error when the redirection fails

Reply:
ok sounds good